src/main/java/com/tikal/jenkins/plugins/multijob/MultiJobBuilder.java

```diff
@@ -393,7 +413,7 @@ else if(!jobStatus.isBuildable() && !phaseConfig.isApplyConditionOnlyIfNoSCMChan
         return true;
     }
 
-    public final class SubJobWorker extends Thread {
+    public final class SubJobWorker implements Callable {

```

Comment:
Now it consumes Jenkins workers instead of spawned threads. IMHO it's a good approach from the scalability PoV